### ISSUE TITLE
Conversation missing field paging

### ISSUE THREAD

**Issue by DerKeil** [restfb/restfb#687]
Hey,

the JSON-object of a conversation-edge is not just returning an array of messages but also two pagination links.

E.g.

```javascript
{
  "id": "t_mid.1460972298060:1e3b607d4aac03ef23",
  "message_count": 181,
  "messages": {
    "data": [
      {
        "created_time": "2017-03-02T09:37:33+0000",
        "id": "m_mid.1488447453538:e313cc8368"
      },
......
    ],
    "paging": {
      "previous": "https://graph.facebook.com/v2.8/t_mid.1460972298060:1e3b607d4aac03ef23/messages?access_token=EAAF4y12rbhoBAHvr433ZAgEpT8ZAXJUDxdVLiD1UZArdt68go7vXj3yMZCM6ZCSS7UxKXFKltHnSSUtVob5Q5aJjfP99cJZCtaIxyxJXRsn7puM0RjWNmVz67jfT2xvDCh2j86B1YZCsldVQpVKYVU0VWuak8AT4CYgtZBB7FSBSx8JeJEukH170C9U1H86YccoZD&limit=25&since=1488447453&__paging_token=enc_AdBu7qlrf6LZAcsThsUNjr6xSqZB2mBhDYRoicz86FYGkjWQyMAgP4tZBJqHSdtf33sFNm1tY3RZAnynX3giBg2DwUbVRd5atNGKuhtwwQuPl1PxKwZDZD&__previous=1",
      "next": "https://graph.facebook.com/v2.8/t_mid.1460972298060:1e3b607d4aac03ef23/messages?access_token=EAAF4y12rbhoBAHvr433ZAgEpT8ZAXJUDxdVLiD1UZArdt68go7vXj3yMZCM6ZCSS7UxKXFKltHnSSUtVob5Q5aJjfP99cJZCtaIxyxJXRsn7puM0RjWNmVz67jfT2xvDCh2j86B1YZCsldVQpVKYVU0VWuak8AT4CYgtZBB7FSBSx8JeJEukH170C9U1H86YccoZD&limit=25&until=1476721922&__paging_token=enc_AdDHfIMLgXk0TSsQkqLiBmGhagAYIPxZCPU2wRNvn6Us2FlZB93rOSNevodDY9Wqb0RguwrdlmKZCAoduihA1Hlk7wpjm6Jpgy7mdpja9PQXtsZAvgZDZD"
    }
  }
}
```

It would be nice to add these two Strings to the `Conversation`. (Similar to the fields `previousPageUrl` and `nextPageUrl` in `Connection`.)

Best
Stefan

**Comment by nbartels** [comment]
yes atm, we don't support these "inner connections". Instead of a list of `Message` objects a `Connection` should be returned so the developer can work with the connection and iterate over all pages. 

But I prefer to use the conversations as entry point to the messages. Because `<t_id>/messages` can used with `FacebookClient#fetchConnection`. To reduce the transferred data, I'd like to set the message limit to 0 in the conversation request. So I get all conversation information without the messages, and fetch the messages on demand.

Nevertheless I think we should do something here, to make it easier.

**Comment by DerKeil** [comment]
Or you fetch the `Conversation` together with the messages, cause you are interested for example in the `message_count`plus all the messages, and in case there are more than 25 messages, then you use the `paging`object contained in the returned `Conversation`to fetch the messages with a `Connection`beginning with the 26th message. So if there is only up to 25 messages at all then you will be fine with only 1 API request, and not with always 2 (1 for the `message_count`and 1 for the messages).

**Comment by nbartels** [comment]
Yes, that's the reason for returning a connection instead of the List. But despite of the conversation a similar approach is needed for comments in a post and in general for all the edges. But at that point there is some consistency missing, but on Facebook side ;)

So I postpone this issue to a 2.x version and try to find a general solution ...

**Comment by DerKeil** [comment]
Eastern European Summer Time (EEST) is 3 hours ahead of Coordinated Universal Time (UTC).

**Comment by ahmedredahussien** [comment]
No it's only 3 hours with some messages but the rest of the messages are 2 hrs ahead

**Comment by nbartels** [comment]
Can you check the date. Perhaps the 2 hours and 3 hours depend on summer/winter time. According to this page (https://www.timeanddate.com/time/zones/eest) in winter the timezone is switched to EET in Europe and then the offset is 2 hours.

**Comment by ahmedredahussien** [comment]
No it's in the same  conversation  and the same running time . I can show you by examples and screen shots give about 7 hr

**Comment by nbartels** [comment]
then please show a list of `rawCreatedTime` and `createdTime` values for the complete conversation. If there are different values in the same conversation, this is interesting and important.

And please check the server time. 

BTW, capture a thread for a different topic is not a good style ...

**Comment by nbartels** [comment]
I close this issue here. The main topic is solved with #1046. The non-topic stuff seems to be solved anyway.
